internal/bminventory/inventory_v2_handlers.go

```diff
@@ -630,6 +630,13 @@ func (b *bareMetalInventory) V2SetIgnoredValidations(ctx context.Context, params
 		return b.setIgnoredValidationsBadRequest("the capability to ignore validations is not available")
 	}
 	cluster, err := b.getCluster(ctx, params.ClusterID.String())
+	preInstallStates := []string{
+		"insufficient", "ready", "pending-for-input",
+	}
+	if !swag.ContainsStrings(preInstallStates, *cluster.Status) {
+		return b.setIgnoredValidationsBadRequest(fmt.Sprintf("cluster %s is not in a pre install state, cannot modify ignored validations", *cluster.ID))
+	}
+
```

Comment:
Use cluster.VerifyClusterUpdatability()

Reply:
This seems reasonable but it does raise some issues when it comes to testing.

The method VerifyClusterUpdatability accepts a cluster, the cluster in the test can be asynchronously updated during the lifespan of the test, meaning that mocks like

`mockClusterApi.EXPECT().VerifyClusterUpdatability(cluster).Return(errors.Errorf("Mock non updatable cluster")).AnyTimes()`

Do not function correcty.

While we are discussing this, I have implemented the mock as 
`mockClusterApi.EXPECT().VerifyClusterUpdatability(gomock.Any()).Return(errors.Errorf("Mock non updatable cluster")).AnyTimes()`

But I feel that this is insufficient. 

Rather, I think we should have a helper method alongside VerifyClusterUpdatability called  VerifyClusterUpdatabiltyByID, which would fetch the cluster from the database and then call VerifyClusterUpdatability, returning the result.

The benefit of this is that VerifyClusterUpdatabiltyByID would be mockable by simply providing a cluster ID and would not be subject to mutations that would prevent the mock from functioning due to async side effects.